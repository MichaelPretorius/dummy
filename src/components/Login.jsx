import React, { useState } from 'react';
import { connect } from 'react-redux';

import FormInput from './FormInput';
import { SubmitButton } from './CustomButton';
import { loginUser } from '../redux/auth/authActions';

const Login = ({ history, loginUser }) => {
  const [credentials, setCredentials] = useState({
    email: '',
    password: '',
  });

  const { email, password } = credentials;

  const handleChange = event => {
    const { name, value } = event.target;
    setCredentials(credentials => ({
      ...credentials,
      [name]: value,
    }));
  };

  const handleSubmit = async event => {
    event.preventDefault();

    await loginUser(email, password, () => {
      setCredentials({
        email: '',
        password: '',
      });
      history.push('/');
    });
  };

  return (
    <div>
      <h2>Log In</h2>
      <form onSubmit={handleSubmit}>
        <FormInput
          required
          type="email"
          label="Email"
          name="email"
          value={email}
          handleChange={handleChange}
        />
        <FormInput
          required
          type="password"
          label="Password"
          name="password"
          value={password}
          handleChange={handleChange}
        />
        <SubmitButton>Log In</SubmitButton>
      </form>
    </div>
  );
};

export default connect(
  null,
  { loginUser }
)(Login);
