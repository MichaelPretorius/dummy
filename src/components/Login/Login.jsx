import React, { useState } from 'react';
import { connect } from 'react-redux';

import { Form, Button, Label, InputField, Header, Wrapper } from '../../styles';
import { loginUser } from '../../redux/auth/authActions';

export const Login = ({ history, loginUser }) => {
  const [credentials, setCredentials] = useState({
    email: '',
    password: '',
  });

  const { email, password } = credentials;

  const handleChange = event => {
    const { name, value } = event.target;
    setCredentials(credentials => ({
      ...credentials,
      [name]: value,
    }));
  };

  const handleSubmit = async event => {
    event.preventDefault();

    await loginUser(email, password, () => {
      setCredentials({
        email: '',
        password: '',
      });
      history.push('/');
    });
  };

  return (
    <>
      <Header center secondary>
        Log In
      </Header>
      <Wrapper>
        <Form onSubmit={handleSubmit} alignEnd>
          <Label secondary>
            Email
            <InputField
              required
              transition
              type="email"
              name="email"
              value={email}
              onChange={handleChange}
            />
          </Label>
          <Label secondary>
            Password
            <InputField
              required
              transition
              type="password"
              name="password"
              value={password}
              onChange={handleChange}
            />
          </Label>
          <Button submit primary>
            Log In
          </Button>
        </Form>
      </Wrapper>
    </>
  );
};

export default connect(
  null,
  { loginUser }
)(Login);
