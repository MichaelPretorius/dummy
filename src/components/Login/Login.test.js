import React from 'react';
import { shallow } from 'enzyme';

import { Login } from './Login';

describe('Login component', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Login />);
  });

  test('should render Login component', () => {
    expect(wrapper.debug()).toMatchSnapshot();
  });
});
