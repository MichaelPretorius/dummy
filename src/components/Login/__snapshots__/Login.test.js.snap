// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component should render Login component 1`] = `
"<div>
  <h2>
    Log In
  </h2>
  <form onSubmit={[Function: handleSubmit]}>
    <FormInput required={true} type=\\"email\\" label=\\"Email\\" name=\\"email\\" value=\\"\\" handleChange={[Function: handleChange]} />
    <FormInput required={true} type=\\"password\\" label=\\"Password\\" name=\\"password\\" value=\\"\\" handleChange={[Function: handleChange]} />
    <SubmitButton>
      Log In
    </SubmitButton>
  </form>
</div>"
`;
